apps/aecore/lib/aecore/naming/tx/name_claim_tx.ex

```diff
@@ -0,0 +1,166 @@
+defmodule Aecore.Naming.Tx.NameClaimTx do
+  @moduledoc """
+  Aecore structure of naming claim.
+  """
+
+  @behaviour Aecore.Tx.Transaction
+
+  alias Aecore.Chain.ChainState
+  alias Aecore.Naming.Tx.NameClaimTx
+  alias Aecore.Naming.Naming
+  alias Aecore.Account.Account
+  alias Aecore.Naming.NameUtil
+  alias Aecore.Account.AccountStateTree
+
+  require Logger
+
+  @typedoc "Expected structure for the Claim Transaction"
+  @type payload :: %{
+          name: String.t(),
+          name_salt: Naming.salt()
+        }
+
+  @typedoc "Structure that holds specific transaction info in the chainstate.
+  In the case of NameClaimTx we have the naming subdomain chainstate."
+  @type tx_type_state() :: ChainState.naming()
+
+  @typedoc "Structure of the Spend Transaction type"
+  @type t :: %NameClaimTx{
+          name: String.t(),
+          name_salt: Naming.salt()
+        }
+
+  @doc """
+  Definition of Aecore NameClaimTx structure
+
+  ## Parameters
+  - name: name to be claimed
+  - name_salt: salt that the name was pre-claimed with
+  """
+  defstruct [:name, :name_salt]
+  use ExConstructor
+
+  # Callbacks
+
+  @spec init(payload()) :: NameClaimTx.t()
+  def init(%{name: name, name_salt: name_salt} = _payload) do
+    %NameClaimTx{name: name, name_salt: name_salt}
+  end
+
+  @doc """
+  Checks name format
+  """
+  @spec validate(NameClaimTx.t()) :: :ok | {:error, String.t()}
+  def validate(%NameClaimTx{name: name, name_salt: name_salt}) do
+    validate_name = NameUtil.normalize_and_validate_name(name)
+
+    cond do
+      validate_name |> elem(0) == :error ->
+        {:error, "#{__MODULE__}: #{validate_name |> elem(1)}: #{inspect(name)}"}
+
+      byte_size(name_salt) != Naming.get_name_salt_byte_size() ->
+        {:error,
+         "#{__MODULE__}: Name salt bytes size not correct: #{inspect(byte_size(name_salt))}"}
+
+      true ->
+        :ok
+    end
+  end
+
+  @spec get_chain_state_name :: Naming.chain_state_name()
+  def get_chain_state_name, do: :naming
+
+  @doc """
+  Claims a name for one account after it was pre-claimed.
+  """
+  @spec process_chainstate(
+          SpendTx.t(),
```

Comment:
Should be NameClaimTx.t() right? I see that it's the same for the other naming transactions as well

Reply:
didn't see this issue from